source/server/admin/admin.cc

```diff
@@ -506,23 +528,114 @@ AdminImpl::addResourceToDump(envoy::admin::v3::ConfigDump& dump,
 }
 
+void AdminImpl::addLbEndpoint(
+    const Upstream::HostSharedPtr& host,
+    envoy::config::endpoint::v3::LocalityLbEndpoints& locality_lb_endpoint) const {
+  auto& lb_endpoint = *locality_lb_endpoint.mutable_lb_endpoints()->Add();
+  if (host->metadata() != nullptr) {
+    lb_endpoint.mutable_metadata()->MergeFrom(*host->metadata());
+  }
+  lb_endpoint.mutable_load_balancing_weight()->set_value(host->weight());
+
+  switch (host->health()) {
+  case Upstream::Host::Health::Healthy:
+    lb_endpoint.set_health_status(envoy::config::core::v3::HealthStatus::HEALTHY);
+    break;
+  case Upstream::Host::Health::Unhealthy:
+    lb_endpoint.set_health_status(envoy::config::core::v3::HealthStatus::UNHEALTHY);
+    break;
+  case Upstream::Host::Health::Degraded:
+    lb_endpoint.set_health_status(envoy::config::core::v3::HealthStatus::DEGRADED);
+    break;
+  default:
+    lb_endpoint.set_health_status(envoy::config::core::v3::HealthStatus::UNKNOWN);
+  }
+
+  auto& endpoint = *lb_endpoint.mutable_endpoint();
+  endpoint.set_hostname(host->hostname());
+  Network::Utility::addressToProtobufAddress(*host->address(), *endpoint.mutable_address());
+  auto& health_check_config = *endpoint.mutable_health_check_config();
+  health_check_config.set_hostname(host->hostnameForHealthChecks());
+  if (host->healthCheckAddress()->asString() != host->address()->asString()) {
+    health_check_config.set_port_value(host->healthCheckAddress()->ip()->port());
+  }
+}
+
+ProtobufTypes::MessagePtr AdminImpl::dumpEndpointConfigs() const {
+  auto endpoint_config_dump = std::make_unique<envoy::admin::v3::EndpointsConfigDump>();
+
+  for (auto& cluster_pair : server_.clusterManager().clusters()) {
+    const Upstream::Cluster& cluster = cluster_pair.second.get();
+    Upstream::ClusterInfoConstSharedPtr cluster_info = cluster.info();
+    envoy::config::endpoint::v3::ClusterLoadAssignment cluster_load_assignment;
+    if (cluster_info->eds_service_name().has_value()) {
+      cluster_load_assignment.set_cluster_name(cluster_info->eds_service_name().value());
+    } else {
+      cluster_load_assignment.set_cluster_name(cluster_info->name());
+    }
+    auto& policy = *cluster_load_assignment.mutable_policy();
+
+    for (auto& host_set : cluster.prioritySet().hostSetsPerPriority()) {
+      policy.mutable_overprovisioning_factor()->set_value(host_set->overprovisioningFactor());
+
+      if (!host_set->hostsPerLocality().get().empty()) {
+        for (int index = 0; index < static_cast<int>(host_set->hostsPerLocality().get().size());
+             index++) {
+          auto locality_host_set = host_set->hostsPerLocality().get()[index];
+
+          if (!locality_host_set.empty()) {
+            auto& locality_lb_endpoint = *cluster_load_assignment.mutable_endpoints()->Add();
+            locality_lb_endpoint.mutable_locality()->MergeFrom(locality_host_set[0]->locality());
```

Comment:
Because `locality_lb_endpoint` has a uniform `locality` as well as `priority`.
`index` is used to get the `locality_lb_endpoint` which includes the hosts with the same locality and priority. However, locality and priority are stored in the `host` but not `locality_lb_endpoint`. Since locality and priority are all the same in hosts of `locality_lb_endpoint`, for here just use the locality and priority of the first host to set the uniform fields.